src/test/java/org/apache/commons/imaging/formats/tiff/TiffRasterDataIntTest.java

```diff
@@ -0,0 +1,203 @@
+                assertEquals(index, test, "Set/get value test failed");
+				instance.setIntValue(x, y, index);
+                test = (int) instance.getIntValue(x, y);
+                assertEquals(index, test, "Set/get int value test failed");
+            }
+        }
+    }
+
+    /**
+     * Test of getValue method, of class TiffRasterData.
+     */
+    @Test
+    public void testGetValue() {
+        for (int y = 0; y < height; y++) {
+            for (int x = 0; x < width; x++) {
+                final int index = y * width + x;
+                int test = (int) raster.getValue(x, y);
+                assertEquals(index, test, "Get into source data test failed at (" + x + "," + y + ")");
+                test = (int) raster.getIntValue(x, y);
+                assertEquals(index, test, "Get into source data test failed at (" + x + "," + y + ")");
+            }
+        }
+    }
+
+    /**
+     * Test of getSimpleStatistics method, of class TiffRasterData.
+     */
+    @Test
+    public void testGetSimpleStatistics_0args() {
+
+        final TiffRasterStatistics result = raster.getSimpleStatistics();
+        assertEquals(0, result.getMinValue(), "Min value failure");
+        assertEquals(width * height - 1, result.getMaxValue(), "Max value failure");
+        assertEquals(meanValue, result.getMeanValue(), "Mean value failure");
+    }
+
+    /**
+     * Test of getSimpleStatistics method, of class TiffRasterData.
+     */
+    @Test
+    public void testGetSimpleStatistics_float() {
+        // exclude the maximum value (width*height-1).  This will result
+        // in a max value of width*height-2
+        final TiffRasterStatistics result = raster.getSimpleStatistics(width * height - 1);
+        assertEquals(width * height - 2, result.getMaxValue(), "Max value failure");
+    }
+
+    /**
+     * Test of getWidth method, of class TiffRasterData.
+     */
+    @Test
+    public void testGetWidth() {
+        assertEquals(width, raster.getWidth(), "Improper width stored");
+    }
+
+    /**
+     * Test of getHeight method, of class TiffRasterData.
+     */
+    @Test
+    public void testGetHeight() {
+        assertEquals(width, raster.getWidth(), "Improper height stored");
+    }
+
+    /**
+     * Test of getData method, of class TiffRasterData.
+     */
+    @Test
+    public void testGetData() {
+        final float[] result = raster.getData();
+		for(int i=0; i<result.length; i++){
+			assertEquals((int)result[i], data[i]);
+		}
+		final int []iResult = raster.getIntData();
+        assertArrayEquals(data, iResult);
+    }
+
+
+     /**
+     * Test of constructors with bad arguments, of class TiffRasterData.
+     */
+    @Test
+    public void testBadConstructor() {
+        try{
+            final TiffRasterData raster = new TiffRasterDataInt(-1, 10);
+            fail("Constructor did not detect bad width");
+        }catch(final IllegalArgumentException illArgEx){
+            // success!
+        }
```

Comment:
@gwlucastrig I believe the tests that are using this pattern `try { something; fail(); } catch {}` is JUnit 4. But for Junit 5 we can use:

```java
assertThrows(IllegalArgumentException.class, () -> new TiffRasterDataInt(-1, 10));
``` 

There are other places in this PR where this can be applied, you can look at GitHub's UI to search for this "fail(" text if you'd like.

Reply:
done